### DISCUSSION TITLE
MagickSettings: probable issue with font & fontPointSize?

### DISCUSSION THREAD

**Discussion by Gounlaf** [dlemstra/Magick.NET#1522] [General]
Looking at ctor of MagicSettings:

https://github.com/dlemstra/Magick.NET/blob/56764717217ce29705c13168cbea815d14073d92/src/Magick.NET/Settings/MagickSettings.cs#L31-L51

`font` & `fontPointSize` are set on the backing property ; but getter&setter use the Drawing object.
So, if the **default value** of `Drawing.Font` and `Drawing.FontPointsize` (defined from `NativeDrawingSettings`) are not the same has default of `NativeMagickSettings`, it could be an issue.

Setting the Drawing property first, then using Font&FontPointsize should be "safer", no?

```csharp
    internal MagickSettings()
    {
        Drawing = new DrawingSettings();

        using var instance = new NativeMagickSettings();
        AntiAlias = instance.AntiAlias;
        BackgroundColor = instance.BackgroundColor;
        ColorSpace = instance.ColorSpace;
        ColorType = instance.ColorType;
        Compression = instance.Compression;
        Debug = instance.Debug;
        Density = CreateDensity(instance.Density);
        Depth = instance.Depth;
        Endian = instance.Endian;
        Extract = MagickGeometry.FromString(instance.Extract);
        Font = instance.Font;
        FontPointsize = instance.FontPointsize;
        Format = EnumHelper.Parse(instance.Format, MagickFormat.Unknown);
        Interlace = instance.Interlace;
        Monochrome = instance.Monochrome;
        Verbose = instance.Verbose;
    }
```

**Comment by dlemstra** [comment]
I suspect that there was probably a good reason to do it like this years ago. But I do think it makes sense to make this more consistent. So feel free to create a PR to change this.

**Reply by Gounlaf** [reply]
:wave: 

What I understand is my _guess_ was true. 

- `NativeMagickSettings.FontPointSize` is 0 by default
- `DrawingSettings.FontPointSize` is 12 by default

So, backed property `_font` & `_fontPointSize` are unsync with DrawingSettings.

Which one should have the primacy?

**Reply by dlemstra** [reply]
Thanks for taking a deep dive into this. We definitely need some unit tests for that then first. And I will also be going to investigate this now. I think that was the reason that I had those backing fields. I will probably not merge your PR then and make the changes myself if I am going to make any changes.

**Reply by Gounlaf** [reply]
Yup. The MR tests are not working because the tests with Affine are not working anymore (because of the font size).

You are definitely more able to do the changes since I don't know the architecture behind 😅

**Reply by dlemstra** [reply]
I was able to change the `Font` field but I must keep the `_fontPointsize` field because it has a different default value.
